### DISCUSSION TITLE
Script to delete unique permissions on all items/docs in a Document Library

### DISCUSSION THREAD

**Discussion by NaveedBajwa** [pnp/powershell#347] [Q&A] [answered]
Is there a script or other way to delete ALL unique permissions across a document library and reset everything to inherit from parent?

**Comment by ToddKlindt** [comment, marked as answer]
I haven't ever done this, but in my head it looks something like this:
```
$ListList = (Get-PnPList -Includes IsSystemList).Where({$_.IsSystemList -EQ $false})
Foreach ($List in $ListList) {
   Get-PnPListItem -List $List | Set-PnPListItemPermission -InheritPermissions -List $List
}
```
Doublecheck it, but that should do it.

**Reply by NaveedBajwa** [reply]
Thanks Todd, 
I am getting the following error

Get-PnPListItem : The attempted operation is prohibited because it exceeds the list view threshold.
At line:3 char:4
+    Get-PnPListItem -List $List | Set-PnPListItemPermission -InheritPe ...
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Get-PnPListItem], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Lists.GetListItem

**Reply by ToddKlindt** [reply]
You have a large list. :) I don't have one handy to test with. Try replacing the line inside of the Foreach loop with this:

`Get-PnPListItem -List $List -PageSize 1000 | Set-PnPListItemPermission -InheritPermissions -List $List
`
That should do it in batches of 1000.

**Reply by NaveedBajwa** [reply]
Legend - Worked like a Charm. Thanks a lot.
